src/main.rs

```diff
@@ -34,6 +46,327 @@ pub mod test_harness;
+    #[behaviour(ignore)]
+    pub orderbook: Orderbook,
+
+    #[behaviour(ignore)]
+    events: VecDeque<NetworkEvent>,
+}
+
+impl Nectar {
+    pub fn new(orderbook: Orderbook) -> Self {
+        Self {
+            orderbook,
+            comit: Comit::default(),
+            events: VecDeque::new(),
+        }
+    }
+
+    fn poll<BIE>(
+        &mut self,
+        _cx: &mut Context<'_>,
+        _params: &mut impl PollParameters,
+    ) -> Poll<NetworkBehaviourAction<BIE, NetworkEvent>> {
+        if let Some(event) = self.events.pop_front() {
+            return Poll::Ready(NetworkBehaviourAction::GenerateEvent(event));
+        }
+
+        // We trust in libp2p to poll us.
+        Poll::Pending
+    }
+}
+
+/// Handle events from the network layer.
+///
+/// These are:
+/// - order expired (once orderbook is added)
+/// - order taken (once orderbook is added)
+/// - swap finalized
+impl libp2p::swarm::NetworkBehaviourEventProcess<comit::network::comit::BehaviourOutEvent>
+    for Nectar
+{
+    fn inject_event(&mut self, event: comit::network::comit::BehaviourOutEvent) {
+        self.events.push_back(NetworkEvent::SwapFinalized(
+            event.local_swap_id,
+            event.remote_data,
+        ))
+    }
+}
+
+#[allow(missing_debug_implementations)]
+pub struct Orderbook;
+
+impl Orderbook {
+    fn publish(&mut self, order: Order) {}
+}
+
+// TODO: Replace this placeholder with something meaningful
+// Representation of an Order - implement mapping/replace with orderbook order
+// once we integrate the orderbook.
+#[derive(Debug, PartialEq)]
+pub struct Order {
+    pub expiration_timestamp: DateTime<Utc>,
+    // should at lest have: trading_pair, position, buy/sell amounts
+    // TODO: decision needed how to depict buy/sell amounts (sat:millicent ...)
+}
+
+pub enum NetworkEvent {
+    // When an order expires a new order is published.
+    // New orders are only published after the old one has expired.
+    // We assume that this event is emitted by the orderbook.
+    OrderExpired(Order),
+
+    // The orderbook registered that somebody wants to take a specific order.
+    // We assume that this event is emitted by the orderbook.
+    OrderTakeRequest(Order), // Emitted by orderbook
+
+    // Message from the network that is emitted at the end of the announce protocol when the
+    // swap is finalized.
+    SwapFinalized(LocalSwapId, RemoteData),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Action {
+    PublishNewOrder(Order),
+    StartSwapExecution(LocalSwapId, RemoteData),
+}
+
+// Bundles the state of the application
+pub struct Maker {
+    btc_balance: bitcoin::Amount,
+    dai_balance: dai::Amount,
+    btc_fee: bitcoin::Amount,
+    dai_fee: dai::Amount,
+    btc_locked_funds: bitcoin::Amount,
+    dai_locked_funds: dai::Amount,
+    btc_max_sell_amount: bitcoin::Amount,
+    dai_max_sell_amount: dai::Amount,
+    rate: MidMarketRate,
+    spread: Spread,
+
+    active_orders: u32,
```

Comment:
Shouldn't this be a container of actual active orders?

Reply:
Most likely yes :) 
It is represented as something very basic at the moment one can take a decision on - it actually remained in there from a previous iteration that implemented a test for an "initial order". 

I will remove it and introduce a proper `Order` store once we actually start building that logic.